### ISSUE TITLE
[pt] Rule “ter nome” → “ter designação”

### ISSUE THREAD

**Issue by marcoagpinto** [languagetool-org/languagetool#6997]
Hello @ricardojosehlima

I have just created a rule that replaces “nome” with “designação”, for academic/scientific usage:
https://github.com/languagetool-org/languagetool/commit/6ebb86cd4883b68c0768a1616a973ec1f1dc6dce

It is a very low-risk rule regarding false positives since it only gave 10 hits in 600 000 sentences:
[0.txt](https://github.com/languagetool-org/languagetool/files/9273910/0.txt)

Basically, it searches for “nome” followed by “de/por/para” and changes “nome” to “designação”.

Next week you will be more free on time and can give your opinion/suggestions regarding this rule.

Notice that in some cases it will require the users to change the gender of the words that follow the change.

❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ 

Thanks!

**Comment by marcoagpinto** [comment]
A philosophical question that is on my mind is if I should always change the “de/por/para” with “por”.

“A equação tem o nome DE técnica quantitativa.” → “A equação é designada POR técnica quantitativa.”

**Comment by marcoagpinto** [comment]
However,

“A equação tem este nome PARA mais poder.”

In this case, which is probably more used in the first rule of the rulegroup, it won't sound too well to change “para” with “por”.

**Comment by ricardojosehlima** [comment]
> A philosophical question that is on my mind is if I should always change the “de/por/para” with “por”.
> 
> “A equação tem o nome DE técnica quantitativa.” → “A equação é designada POR técnica quantitativa.”

I agree!

**Comment by ricardojosehlima** [comment]
> Hello @ricardojosehlima
> 
> I have just created a rule that replaces “nome” with “designação”, for academic/scientific usage: [6ebb86c](https://github.com/languagetool-org/languagetool/commit/6ebb86cd4883b68c0768a1616a973ec1f1dc6dce)
> 
> It is a very low-risk rule regarding false positives since it only gave 10 hits in 600 000 sentences: [0.txt](https://github.com/languagetool-org/languagetool/files/9273910/0.txt)
> 
> Basically, it searches for “nome” followed by “de/por/para” and changes “nome” to “designação”.
> 
> Next week you will be more free on time and can give your opinion/suggestions regarding this rule.
> 
> Notice that in some cases it will require the users to change the gender of the words that follow the change.
> 
> heart heart heart heart heart heart heart heart
> 
> Thanks!

It's a good idea and the hits worked well!
Congratulations for +1 rule!

**Comment by marcoagpinto** [comment]
@ricardojosehlima 

Dear Ricardo,

I have rewritten the rule:
https://github.com/languagetool-org/languagetool/commit/62173f444f7faf283c9cc0b3f3751cd234862ae7

This is what I based myself on to create it:
```
RULE1:
A equação tem o nome DE técnica quantitativa. → A equação é designada POR técnica quantitativa.
As equações têm o nome DE técnica quantitativa. → As equações são designadas POR técnica quantitativa.
O projeto tem o nome DE técnica quantitativa. → O projeto é designado POR técnica quantitativa.
Os projetos têm o nome DE técnica quantitativa. → Os projetos são designados POR técnica quantitativa.

RULE2:
O projeto tem este nome por ser impressionante. → O projeto tem esta designação por ser impressionante.
Os projetos têm este nome por ser impressionante. → Os projetos têm esta designação por ser impressionante.

RULE3:
O projeto é assim chamado por ser impressionante. → O projeto é assim designado por ser impressionante.
Os projetos são assim chamados por serem impressionantes. → Os projetos são assim designados por serem impressionantes.
A equação é assim chamada por ser brilhante. → A equação é assim designada por ser brilhante.
As equações são assim chamadas por serem brilhantes. → As equações são assim designadas por serem brilhantes.
```

And here are the new results in 600 000 sentences:
[1_new.txt](https://github.com/languagetool-org/languagetool/files/9302121/1_new.txt)

Is all this properly done?

If so, we can move on to other “adventures” 😄 😄 😄 😄 😄 😄 😄 😄 😄 😄 😄 😄 😄 

Thanks!
